pkg/volume/plugins.go

```diff
@@ -121,6 +122,9 @@ type VolumeHost interface {
 	// the provided spec.  See comments on NewWrapperBuilder for more
 	// context.
 	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
+
+	//Get cloud provider from kubelet
+	GetCloudProvider() cloudprovider.Interface
```

Comment:
why can't this instantiate openstack directly?

On Wed, Aug 19, 2015 at 9:02 PM, spothanis <notifications@github.com> wrote:

> In pkg/volume/plugins.go
> <https://github.com/kubernetes/kubernetes/pull/6689#discussion_r37493239>:
>
> > @@ -121,6 +122,9 @@ type VolumeHost interface {
> >  	// the provided spec.  See comments on NewWrapperBuilder for more
> >  	// context.
> >  	NewWrapperCleaner(spec *Spec, podUID types.UID, mounter mount.Interface) (Cleaner, error)
> > +
> > +	//Get cloud provider from kubelet
> > +	GetCloudProvider() cloudprovider.Interface
>
> @saad-ali <https://github.com/saad-ali> openstack couldn't instantiate
> itself, so I submitted a different patch to make kubelet cloudprovider
> aware. How ever without adding it to the volume host, the only other way
> was to pass it into the builder, which also didn't seem correct.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/6689/files#r37493239>.
>